src/java/org/apache/cassandra/tcm/log/LogReader.java

```diff
@@ -110,6 +112,54 @@ default LogState getLogState(Epoch startEpoch)
         }
     }
 
+    default LogState getLogState(Epoch start, Epoch end)
```

Comment:
should probably override this method in `NoOpLogStorage`

Reply:
Should there be a way for this to not return a snapshot (they can get fairly large)? I'm thinking of a situation when reconstructing the log due to gaps in the log, then we could just reconstruct from the current local epoch, with no need to get a snapshot from the CMS

Reply:
Technically, we only need accord topologies. Maybe we could have some form of a lagging index of Accord Topologies available via CMS?